test/e2e/framework/service/jig.go

```diff
@@ -1043,6 +1043,23 @@ func (j *TestJig) CreateServicePods(c clientset.Interface, ns string, replica in
 	framework.ExpectNoError(err, "Replica must be created")
 }
 
+// CreateTCPUDPServicePods creates a replication controller with the label same as service. Service listens to TCP and UDP.
+func (j *TestJig) CreateTCPUDPServicePods(c clientset.Interface, ns string, replica int) {
+	config := testutils.RCConfig{
+		Client:       c,
+		Name:         j.Name,
+		Image:        framework.ServeHostnameImage,
+		Command:      []string{"/agnhost", "serve-hostname", "--http=false", "--tcp", "--udp"},
```

Comment:
nit: I am wondering we can merge this new function into the above existing function because the difference is just this command here. But at this time, it is fine and we can take care later.

Reply:
Thanks for the suggestion, I was in the same mindset to have single function and leverage flags as args but to keep PR simple, I have not introduced here. As you mentioned, I can create another PR to refactor it.

Following is raw code.
```
var ServeHostNameImageDefaultFlags = map[string]string{
	HTTP: "--http",
	TCP:  "--tcp",
	UDP:  "--udp",
}


func enableServeHostNameImageCmdFlags(enableFlags ...string) (enabledFlags []string){
	gomega.Expect(len(enableFlags)).NotTo(gomega.BeNumerically(">", 2), "HTTP only or TCP/UDP requests can be served but not both simultaneously")
	var flags = make(map[string]string,len(ServeHostNameImageDefaultFlags))
	for _,f := range enableFlags{
		switch f{
		case TCP, UDP:
			framework.ExpectNotEqual(flags[f],ServeHostNameImageDefaultFlags[HTTP], "HTTP flag can not be enabled along with TCP/UDP")
			flags[f] = "--http=false"
			flags[f] = ServeHostNameImageDefaultFlags[f]
		case HTTP:
			if flags[TCP] != "" || flags[UDP] != ""{
				framework.Failf("TCP/UDP flag is enabled so HTTP flag can not be enabled")
			}
			flags[f] = ServeHostNameImageDefaultFlags[f]
		default:
			framework.Failf("Unexpected protocol flags are passed. Valid protocol flags are HTTP, TCP and UDP")
		}
	}
	for _, f := range flags{
                enabledFlags = append(enableFlags, f)
	}
    return
}

flags := enableServeHostNameImageCmdFlags(TCP, UDP)
```

Reply:
@mgdevstack could you please create a follow up issue for this, so we dont forget?
once we get that we can cancel the hold on this.